pkg/controller/control.go

```diff
@@ -759,6 +765,31 @@ func (c *Controller) WriteAt(b []byte, off int64) (int, error) {
 	return n, err
 }
 
+func (c *Controller) writeInWOMode(b []byte, off int64) (int, error) {
+	sectorSize := int64(4096)
+	bufLen := int64(len(b))
+	// buffer b is sectorSize aligned
+	if (bufLen >= sectorSize) && (off%sectorSize == 0) && ((off+bufLen)%sectorSize == 0) {
+		return c.backend.WriteAt(b, off)
+	}
+
+	readOffsetStart := (off / sectorSize) * sectorSize
+	readOffsetEnd := (((off + bufLen) / sectorSize) + 1) * sectorSize
+	readBuf := make([]byte, readOffsetEnd-readOffsetStart)
+	if _, err := c.backend.ReadAt(readBuf, readOffsetStart); err != nil {
+		return 0, err
```

Comment:
Suggest an enriching message with this err like `failed to retrieve aligned sectors from RW replicas..`.

Reply:
Thank you. Fixed